test/extensions/filters/http/geoip/config_test.cc

```diff
@@ -194,19 +119,16 @@ TEST(GeoipFilterConfigTest, GeoipFilterConfigMissingProvider) {
 TEST(GeoipFilterConfigTest, GeoipFilterConfigUnknownProvider) {
   TestScopedRuntime scoped_runtime;
   DummyGeoipProviderFactory dummy_factory;
-  Registry::InjectFactory<GeoipProviderFactory> registered(dummy_factory);
+  Registry::InjectFactory<Geolocation::GeoipProviderFactory> registered(dummy_factory);
   std::string filter_config_yaml = R"EOF(
-    geo_headers_to_add:
-      country: "x-geo-country"
-      region: "x-geo-region"
     provider:
         name: "envoy.geoip_providers.unknown"
   )EOF";
   GeoipFilterConfig filter_config;
 
   TestUtility::loadFromYaml(filter_config_yaml, filter_config);
   NiceMock<Server::Configuration::MockFactoryContext> context;
-  EXPECT_CALL(context, messageValidationVisitor());
+  // EXPECT_CALL(context, messageValidationVisitor());
```

Comment:
Is this `TODO(): fix the test that is not behaving as expected` or this line should just be removed as no longer relevant or `TODO(): after some later feature this expectation will be valid`?

Reply:
this is another TODO: this line is never be hit in codepath because the code will terminate with exception earlier while trying to create provider with unknown type. I will remove this line